coral/src/app/features/requests/acls/AclRequests.tsx

```diff
@@ -79,29 +81,31 @@ function AclRequests() {
 
   const [errorMessage, setErrorMessage] = useState("");
 
-  const {
-    isLoading: deleteIsLoading,
-    mutate: deleteRequest,
-    isSuccess: deleteIsSuccess,
-  } = useMutation({
+  const { isLoading: deleteIsLoading, mutate: deleteRequest } = useMutation({
     mutationFn: deleteAclRequest,
-    onSuccess: (responses) => {
+    onSuccess: async (responses) => {
       const response = responses[0];
 
       if (response.result !== "success") {
-        return setErrorMessage(
-          response.message || response.result || "Unexpected error"
-        );
+        closeModal();
+        throw Error(response.message || response.result || "Unexpected error");
       }
 
       setErrorMessage("");
-      setModals({ open: "DETAILS", req_no: null });
 
       // We need to refetch all aclrequests queries to keep Table state in sync
-      queryClient.refetchQueries(["aclRequests"]);
+      // We await it to only close modal after refetch is done
+      await queryClient.refetchQueries(["aclRequests"]);
+
+      closeModal();
     },
     onError: (error: Error) => {
-      setErrorMessage(parseErrorMsg(error));
+      // Case when error is from the server
```

Comment:
the modal needs to be closed here, too. 
Also - the error from `throw Error` in line 91 will be caught here, so it's enough to do `closeModal` here.

Reply:
Does it make a difference where we close the modal? I don't object to putting it in `onError`, but as I see it it has no impact on the flow, and I'm curious to know of a reason why it would have one ^^

Reply:
>Does it make a difference where we close the modal? 
🤔  if there is a real error returned (not our fake one that comes as a `response` to us) we would never reach the `closeModal` in `onSuccess` and it wouldn't close? Do I have a 🍝  in my thinking about that? Wait, let me check :D

Reply:
That is a good reason I did not think of, thank you ^^